gridt/src/app/settings/change-bio/change-bio.page.ts

```diff
@@ -0,0 +1,73 @@
+import { Component, Input } from '@angular/core';
+import { PopoverController, LoadingController, AlertController } from '@ionic/angular';
+
+import { Observable, timer } from 'rxjs';
+import { timeout } from 'rxjs/operators';
+
+import { NgForm } from '@angular/forms';
+
+import { SettingsService} from '../../core/settings.service'
+import { Settings } from '../../core/settings.model';
+import { ApiService } from '../../core/api.service'
+
+@Component({
+  selector: 'app-change-bio',
+  templateUrl: './change-bio.page.html',
+  styleUrls: ['./change-bio.page.scss'],
+})
+export class ChangeBioPage{
+  edit_bio$: boolean = true;
+
+  constructor(
+    private SetService: SettingsService,
+    private alertCtrl: AlertController,
+    private loadingCtrl: LoadingController,
+    public popoverCntrl: PopoverController,
+    private api: ApiService
+  ) { }
+
+  ngOnInit() {
+  }
+
+  public closePopover() {
+    this.popoverCntrl.dismiss({
+      'dismissed': true
+    });
+  }
+
+  /*
+   * Change the bio
+   */
+  async write_bio(form: NgForm) {
+    this.edit_bio$ = true;
+
+    const el = await this.loadingCtrl.create({
+      message: 'Updating bio...'
+    });
+
+    el.present();
+
+    this.api.putBio$(form.value.bio).subscribe(
+      () => {
+        el.dismiss();
+        this.closePopover();
+      },
+      (error) => {
+        el.dismiss();
+        this.showError(error);
+      }
+    );
+
+    timer(500).subscribe( () => this.SetService.updateUserSettings());
```

Comment:
Why the .5 second wait?

Reply:
If you want to make sure that the server settings are the same as those stored locally, subscribing to the request makes more sense. If that doesn't error, you can be pretty sure that it has arrived properly, if you still want to check it would make sense to do the request at that point, but .5 seconds is pretty arbitrary, especially on phones that do not always have the best connections.